include/unicorn/unicorn.h

```diff
@@ -248,6 +248,8 @@ typedef enum uc_hook_type {
 // Hook type for all events of illegal memory access
 #define UC_HOOK_MEM_INVALID (UC_HOOK_MEM_UNMAPPED + UC_HOOK_MEM_PROT)
 // Hook type for all events of valid memory access
+// NOTE: UC_HOOK_MEM_READ is triggered before UC_HOOK_MEM_READ_PROT and UC_HOOK_MEM_READ_UNMAPPED, so
+//       this hook will technically trigger on some invalid reads.
```

Comment:
"this hook **will** technically trigger on some invalid reads" -> do you want to change "will" to "may" ?